homeassistant/components/group/climate.py

```diff
@@ -0,0 +1,342 @@
+                entities,
+                config_entry.options.get(
+                    CONF_TEMPERATURE_UNIT, hass.config.units.temperature_unit
+                ),
+            )
+        ]
+    )
+
+
+class ClimateGroup(GroupEntity, ClimateEntity):
+    """Representation of a climate group."""
+
+    _attr_available: bool = False
+    _attr_assumed_state: bool = True
+
+    def __init__(
+        self,
+        unique_id: str | None,
+        name: str,
+        entity_ids: list[str],
+        temperature_unit: str,
+    ) -> None:
+        """Initialize a climate group."""
+        self._entity_ids = entity_ids
+
+        self._attr_name = name
+        self._attr_unique_id = unique_id
+        self._attr_extra_state_attributes = {ATTR_ENTITY_ID: entity_ids}
+
+        self._attr_temperature_unit = temperature_unit
+
+        # Set some defaults (will be overwritten on update)
+        self._attr_supported_features = ClimateEntityFeature(0)
+        self._attr_hvac_modes = [HVACMode.OFF]
+        self._attr_hvac_mode = None
+        self._attr_hvac_action = None
+
+        self._attr_swing_modes = None
+        self._attr_swing_mode = None
+
+        self._attr_fan_modes = None
+        self._attr_fan_mode = None
+
+        self._attr_preset_modes = None
+        self._attr_preset_mode = None
+
+    @callback
+    def async_update_group_state(self) -> None:
+        """Query all members and determine the climate group state."""
+        self._attr_assumed_state = False
+
+        states = [
+            state
+            for entity_id in self._entity_ids
+            if (state := self.hass.states.get(entity_id)) is not None
+        ]
+        self._attr_assumed_state |= not states_equal(states)
+
+        # Set group as unavailable if all members are unavailable or missing
+        self._attr_available = any(state.state != STATE_UNAVAILABLE for state in states)
+
+        # Temperature settings
+        self._attr_target_temperature = reduce_attribute(
+            states, ATTR_TEMPERATURE, reduce=lambda *data: mean(data)
+        )
+
+        self._attr_target_temperature_step = reduce_attribute(
+            states, ATTR_TARGET_TEMP_STEP, reduce=max
+        )
+
+        self._attr_target_temperature_low = reduce_attribute(
+            states, ATTR_TARGET_TEMP_LOW, reduce=lambda *data: mean(data)
+        )
+        self._attr_target_temperature_high = reduce_attribute(
+            states, ATTR_TARGET_TEMP_HIGH, reduce=lambda *data: mean(data)
+        )
+
+        self._attr_current_temperature = reduce_attribute(
+            states, ATTR_CURRENT_TEMPERATURE, reduce=lambda *data: mean(data)
+        )
+
+        self._attr_min_temp = reduce_attribute(states, ATTR_MIN_TEMP, reduce=max)
+        self._attr_max_temp = reduce_attribute(states, ATTR_MAX_TEMP, reduce=min)
+        # End temperature settings
+
+        # available HVAC modes
+        all_hvac_modes = list(find_state_attributes(states, ATTR_HVAC_MODES))
```

Comment:
Should the group only support HVAC that are available for all climate entities?

Reply:
Done